### DISCUSSION TITLE
Use the sensorbox for charging current

### DISCUSSION THREAD

**Discussion by Xnygra** [serkri/SmartEVSE-3#113] [Q&A] [answered]
I am using the sensorbox, but now my grid company installed a new smartmeter that give me the phase currents on the P1 port, so I can use this instead of the sensorbox. 
So, could i somehow make use of my sensorbox for measuring the charge currents instead?

**Comment by HA-TB303** [comment]
Yes you can, using esphome and an esp8266 or esp32 based DSMR interface:

```
interval:
  - interval: 5sec
    then:
            - http_request.post: 
                url: !lambda |-
                    std::string url;
                    url.append("http://192.168.207.121/currents?L1=");
                    url.append(to_string(id(currentl1).state *10));
                    url.append("&L2=");
                    url.append(to_string(id(currentl2).state *10));
                    url.append("&L3=");
                    url.append(to_string(id(currentl3).state *10));
                    return url.c_str();

```
You need to modify your sensors (adding the id's). The IP-adres is that of your smartevse v3 running serkri.

**Reply by Xnygra** [reply]
I already got a ESP P1 reader setup and get the data into the SmartEvse, my question is if I can (re-)use the SmartEvse sensor box for measuring the charging current (attaching the clamps to wires from the contactor) and use it for measuring the charging currents, instead of the incoming mains.

**Reply by HA-TB303** [reply]
I totally misunderstood your question, so just ignore my answer ;)
I have no idea about the sensorbox, I am trying to avoid using that thing.

**Comment by dingo35** [comment]
The Sensorbox was originally designed to measure the MainsMeter currents; the  SmartEVSE firmware currently limits the sensorbox being used for the MainsMeter only, I would be happy to remove that constraint for you; from reading the code I would think nothing strange would happen but since I cannot test this myself (no sensorbox here) it is, you should test it thoroughly before I can make that change mainstream.
Are you willing and able to do that?

**Reply by Xnygra** [reply]
Hi Dingo, Yes, very happy to test it. Not a coder my self, but if you can give me a compiled version, I can update my SmartEvse and try it. 
How would you go about it, have an option in the GUI settings what to use the Sensor box for?

**Comment by dingo35** [comment]
After flashing, you would configure (at the LCD screen) your MainsMeter to EM_API (so it will receive your P1 measurments from your esphome firmware) and your EVMeter to Sensorbox. 
[eastron1p+empty-slots+sensorbox-on-evmeter.zip](https://github.com/serkri/SmartEVSE-3/files/10969135/eastron1p%2Bempty-slots%2Bsensorbox-on-evmeter.zip)

**Reply by Xnygra** [reply]
Got the firmware flashed and set MainsMeter to API and EVMeter to Sensorbox, but all zeros on the EVMeter currents, EV meter description: Sensorbox. 
![Screendump](https://user-images.githubusercontent.com/44607420/225126073-fbe1435a-00ee-4fbf-b4b9-4b50576257bb.JPG)

**Comment by dingo35** [comment]
Ok thx for the feedback, Iooks like you got it all right, Ill look into it.

**Comment by dingo35** [comment]
Ok I made some changes so now you have to configure some grid/call stuff (as you did have to do when using the sensorbox as mainsmeter entry) when select evmeter=sensorbox. 

But I don't think this will solve the problem; to investigate, after flashing this version, telnet to your device and select v(verbose), and please post your debug log here so we can see whats going on on your modbus.
[eastron1p+empty-slots+sensorbox-on-evmeter2-debug.zip](https://github.com/serkri/SmartEVSE-3/files/10974902/eastron1p%2Bempty-slots%2Bsensorbox-on-evmeter2-debug.zip)

**Reply by Xnygra** [reply]
I would like to, but Telnet to my device is above my knowledge, sorry. But please point me in right direction and ill try my best to get you the debug.

**Comment by dingo35** [comment]
Just enter:
telnet ip-address-of-your-smartevse 

on the command line of your operating system. It will connect and show you a screen with possible commands;press "v" and then <enter>.
You will see debug message scrolling om your screen.

Make sure your evse is charging when you do all this.
Then copy/paste a page or so of those messages to this website....

**Reply by Xnygra** [reply]
Thank you for the explanation, got the log. 

As you said, still nothing under EV Meter, all zeros. 

But I got to think that this idea with the  sensor box as EV Meter might not be so great as its only a Amp meter and not a Kwh meter, that you really want to have. You can do some estimated calculation to get kwh, but they will not be correct as you don't have the measured voltages over the phases. 
[SmartEvseLog.txt](https://github.com/serkri/SmartEVSE-3/files/11002354/SmartEvseLog.txt)

**Comment by dingo36** [comment, marked as answer]
Ok I can't see anything strange going on there, the modbus request is being made to the sensorbox and it answers with a valid modbus response.... 
You are right, the use is limited because it will not give you power and/or energy usage....

So I guess not spending any more time on this.....
